### PR TITLE
feat: added hidePricing prop to GlobalFooter, GlobalHeader

### PR DESCRIPTION
**Pull request by JoshuaKGoldberg**
### Overview

Adds a `hidePricing` prop to hide the _Pro Pricing_ link.

This will be used by our PWA when in ChromeOS mode to hide the ability to see about monthly plans, since the ChromeOS PWA will only allow yearly plans _(for now)_.

### Notes

Refactors the tests for the header to use `@codecademy/gamut-tests` and not set up views in `beforeEach` blocks.

Eventually I'd like to ideate on setting up some kind of context so we don't have to prop drill... maybe later 😄 

### PR Checklist

- ~[ ] Related to designs:~
- [x] Related to JIRA ticket: WEB-1569
- [x] I have run this code to verify it works
- [x] This PR includes unit tests for the code change

### CODE REVIEW

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6116cfa79b555600b874fd57--gamut-preview.netlify.app>https://6116cfa79b555600b874fd57--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6116cfa79b555600b874fd57>Deploy Logs</a></p>

File: packages/gamut-labs/src/GlobalHeader/__tests__/GlobalHeader-test.tsx [hunk 1]
```diff
@@ -116,252 +114,263 @@ const renderElementProps: GlobalHeaderProps = {
...
+    it('renders pricingDropdown when hidePricing is falsy', () => {
+      const { view } = renderView(anonHeaderProps);
+      view.getAllByText(pricingDropdown.text);
     });
 
-    test('plansPricingDropdown', () => {
-      screen.getAllByText(pricingDropdown.text);
+    it('does not render pricingDropdown when hidePricing is true', () => {
+      const { view } = renderView({ ...anonHeaderProps, hidePricing: true });
+      expect(view.queryByText(pricingDropdown.text)).toBeFalsy();
```
Comment (jakemhiller) [inline packages/gamut-labs/src/GlobalHeader/__tests__/GlobalHeader-test.tsx:152]: hm, `setupRtl` should probably expose `screen`.

Comment (jakemhiller) [reply packages/gamut-labs/src/GlobalHeader/__tests__/GlobalHeader-test.tsx:152]: i guess that would be weird. I do miss using it when i use our utils.

**Review by jakemhiller — APPROVED** [review]

File: packages/gamut-labs/src/GlobalHeader/__tests__/GlobalHeader-test.tsx (hunk 1 above)
Comment (JoshuaKGoldberg) [reply packages/gamut-labs/src/GlobalHeader/__tests__/GlobalHeader-test.tsx:152]: Ha, I don't... But in theory you can still use it.

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.5.34-alpha.812092.0<br/>@codecademy/gamut-labs@19.21.2-alpha.812092.0<br/>@codecademy/styleguide@43.7.2-alpha.812092.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://611a89b48cbddd00e71a88c9--gamut-preview.netlify.app>https://611a89b48cbddd00e71a88c9--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/611a89b48cbddd00e71a88c9>Deploy Logs</a></p>

**Comment by saghdaey** [comment]
testing in the monolith preview 
- Q: should we hide the footer pricing links? (can be a separate pr)
- Q: should free users see the "upgrade to pro" button in the header?
- 
i'm logged in as a free user below:
Screen Shot 2021-08-16 at 12 54 10 PM

**Comment by JoshuaKGoldberg** [comment]
> Should we hide the footer pricing links? (can be a separate pr)

Great callout! For ease of integration, I added that here in bfdd1df

> Should free users see the "upgrade to pro" button in the header?

Yes they should. The checkout page is different and only supports yearly plans, hence the removal of the full pricing page links

**Review by saghdaey — APPROVED** [review]
lgtm:)

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.5.34-alpha.72b379.0<br/>@codecademy/gamut-labs@19.21.2-alpha.72b379.0<br/>@codecademy/styleguide@43.7.2-alpha.72b379.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://611aa197dca8042144ee1f26--gamut-preview.netlify.app>https://611aa197dca8042144ee1f26--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/611aa197dca8042144ee1f26>Deploy Logs</a></p>

**Review by katiezutter — APPROVED** [review]
All looks good! Thanks so much for updating tests 👍

**Review by dreamwasp — APPROVED** [review]
✅

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.5.34-alpha.296d8d.0<br/>@codecademy/gamut-labs@19.21.2-alpha.296d8d.0<br/>@codecademy/styleguide@43.7.2-alpha.296d8d.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://611acc5d505a3627213aa4f9--gamut-preview.netlify.app>https://611acc5d505a3627213aa4f9--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/611acc5d505a3627213aa4f9>Deploy Logs</a></p>

**Comment by JoshuaKGoldberg** [comment]
Btw I added just a bit of padding above _Enterprise Plans_ for `hidePricing` so that it's even. By default it's slightly above _Support_.

Screenshot of Enterprise Plans and Support side-by-side and even in a footer

**Pull request merged**